config/docs/site_processing_config_README.md

```diff
@@ -0,0 +1,211 @@
+e.g.*  
+    - Barcode  
+    - Location  
+    - Count  
+    - Resize *(Resized images are used for visual output and not measurements.)*
+
+## process-spots:
+
+`perform:` *Do you want to perform 1.process-spots step?  
+Set to* true *or* false.
+
+`output_basedir:` *Name the folder where you would like this step to output files.  
+e.g.* data
+
+`barcode_cols:` *The column that contains the barcodes (or columns that contain the barcodes) assigned to a given spot.  
+e.g.*  
+    - Barcode_MatchedTo_Barcode
+
+`gene_cols:` *The column that contains the gene (or columns that contain the genes) assigned to a given spot.  
+e.g.*  
+    - Barcode_MatchedTo_GeneCode
+
+`location_cols:` *The columns that define the location of each spot (generally X and Y coordinates).  
+e.g.*  
+    - Location_Center_X  
+    - Location_Center_Y
+
+`spot_score_cols:` *The column that contains the match score for the barcode (or columns that contain the match score for the barcodes) assigned to a spot.  
+e.g.*  
+    - Barcode_MatchedTo_Score
+
+`foci_cols:`*The columns that contain the top match for barcode and corresponding gene.
+e.g.*  
+    - Barcode_BarcodeCalled  
+    - Barcode_MatchedTo_ID
+
+`cell_filter:`*Select the categories of cells that you wish to use for analysis. Categories are defined in cell_quality_utils.py.  
+Make sure that the options you set here are available in the categorize_cells method that you set above in core config.  
+Typical e.g.*  
+    - Perfect  
+    - Great  
+*Other possible options e.g.*  
+    - Imperfect *(only in simple)*  
+    - Imperfect-High *(only in simple simple_plus)*  
+
+## process-cells:
+
+`perform:` *Do you want to perform 2.process-cells step?  
+Set to* true *or* false.
+
+`prefilter-features:` *Do you want to use the prefiltering that you created in step 0?  
+Set to* true *or* false.
+
+`output_basedir:` *Name the folder where you would like this step to output files.  
+e.g.* data
+
+`sort_col:` *Column used for sorting data numerically.  
+e.g.* Metadata_Cells_ObjectNumber  
+
+`merge_columns:`
+*CellProfiler outputs measurements for each compartment in a separate .csv.  
+In this step we merge them together using the merge_columns.*  
++ `image_column:` *The name of the column that describes the image number.  
+e.g.* ImageNumber  
++`linking_compartment:` *The compartment used to link other compartments.  
+e.g.* Cytoplasm  
++`linking_columns:`*Column names that link your parent compartment to other compartments.  
+You will need to set a column name for each compartment set in core: compartments.  
+The column names listed, must also exist in the data for the compartment listed in `linking_compartment`.*   
+  - `cells:` *e.g.* Metadata_Cytoplasm_Parent_Cells  
+  - `nuclei:` *e.g.* Metadata_Cytoplasm_Parent_Nuclei
++ `metadata_linking_columns:` *Column names that link ...   
+e.g.*
+- Metadata_Foci_site  
+- Metadata_Cells_ObjectNumber  
```

Comment:
```suggestion
`merge_columns:`
*CellProfiler outputs measurements for each compartment in a separate .csv.  
In this step we merge them together using the merge_columns.*
- `image_column:` *The name of the column that describes the image number.  
e.g.* ImageNumber  
- `linking_compartment:` *The compartment used to link other compartments.  
e.g.* Cytoplasm  
- `linking_columns:`*Column names that link your parent compartment to other compartments.  
You will need to set a column name for each compartment set in core: compartments.  
The column names listed, must also exist in the data for the compartment listed in `linking_compartment`.*   
  - `cells:` *e.g.* Metadata_Cytoplasm_Parent_Cells  
  - `nuclei:` *e.g.* Metadata_Cytoplasm_Parent_Nuclei
- `metadata_linking_columns:` *Column names linking spot metadata with single cell morphology readouts  
e.g.* Metadata_Foci_site, Metadata_Cells_ObjectNumber  
```

I think this suggested edits addresses the outstanding problems:

> Switching to the bulleting for items with two levels creates a problem if the lowest level is a list of items because it is automatically formatting the - item with a bullet.
e.g. process-cells: merge_columns: linking_columns
Do you know a formatting trick to get around this?

I don't think we need to mirror the formatting in the real `yaml` file. I imagine that once a user sees the default `yaml` structure, this README file will serve to specifically answer their documentation questions. Maybe it is worth adding a reference to the [YAML wikipedia](https://en.wikipedia.org/wiki/YAML)? and a note that "structure = function" in the YAML file 😃 

> Can you write the explanation for process-cells: merge_columns: metadata_linking_columns? I'm having a hard time putting it to words.

This is a good point - I also had a bit of trouble. Are we actually using this in the preprocessing module? I know we're using it in the generate profiles module, but maybe I am overlooking something. It is possible that my addition (paste again below) is only relevant for the profiling section. If this is the case, we should either 1) drop the options or 2) rename the argument something that is more meaningful.

```
`metadata_linking_columns:` *Column names linking spot metadata with single cell morphology readouts
```

What do you think?

Reply:
Can you confirm this to make sure I'm reading things properly?
In 2.process-cells, the `metadata_linking_columns` is a part of the `merge_info` that is passed to `merge_single_cell_compartments`  in `paint_utils`.
But `merge_single_cell_compartments` only uses the first three variables of `merge_info` ( `image_column`, `linking_compartment`, and `linking_columns`) so we can in fact remove `metadata_linking_columns`.

Reply:
> Can you confirm this to make sure I'm reading things properly?

Confirmed. We only use `metadata_linking_columns` in the `1.generate-profiles` module.